generic3g/GenericConfig.F90

```diff
@@ -1,14 +1,13 @@
 module mapl3g_GenericConfig
-   use esmf, only: Esmf_Config
-   use yaFyaml, only: YAML_Node
+   use esmf, only: Esmf_HConfig, ESMF_Config
    implicit none
    private
   
    public :: GenericConfig
   
    type :: GenericConfig
       type(ESMF_Config), allocatable :: esmf_cfg
-      class(YAML_Node), allocatable :: yaml_cfg
+      type(ESMF_HConfig), allocatable :: yaml_cfg
```

Comment:
Does not need to be fixed in this PR, but I think we can now just eliminate GenericConfig.   But maybe not just yet.